toolchain/base/value_store.h

```diff
@@ -44,7 +44,11 @@ struct IdTag {
          // doesn't collide with anything else (though with the
          // second-highest-bit-tagging this might not be needed).
         id_tag_(llvm::reverseBits((((id_index + 1) << 1) | 1) << 1)),
-        initial_reserved_ids_(initial_reserved_ids) {}
+        initial_reserved_ids_(initial_reserved_ids) {
+    CARBON_DCHECK(
```

Comment:
```suggestion
    CARBON_CHECK(
```
we use check unless there is a perf concern